RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,609 @@
+
+#include "TrackingTools/GeomPropagators/interface/StraightLinePlaneCrossing.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+//
+// class declaration
+//
+struct DeepCoreCache {
+  mutable tensorflow::GraphDef* graph_def;
+};
+
+class DeepCoreSeedGenerator : public edm::stream::EDProducer<edm::GlobalCache<DeepCoreCache>> {
+public:
+  explicit DeepCoreSeedGenerator(const edm::ParameterSet&, const DeepCoreCache*);
+  ~DeepCoreSeedGenerator() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  // A pointer to a cluster and a list of tracks on it
+
+  // static methods for handling the global cache
+  static std::unique_ptr<DeepCoreCache> initializeGlobalCache(const edm::ParameterSet&);
+  static void globalEndJob(DeepCoreCache*);
+
+  struct TrackAndState {
+    TrackAndState(const reco::Track* aTrack, TrajectoryStateOnSurface aState) : track(aTrack), state(aState) {}
+    const reco::Track* track;
+    TrajectoryStateOnSurface state;
+  };
+
+  template <typename Cluster>
+  struct ClusterWithTracks {
+    ClusterWithTracks(const Cluster& c) : cluster(&c) {}
+    const Cluster* cluster;
+    std::vector<TrackAndState> tracks;
+  };
+
+  typedef ClusterWithTracks<SiPixelCluster> SiPixelClusterWithTracks;
+
+  // typedef std::vector<SiPixelClusterWithTracks> SiPixelClustersWithTracks;
+
+  double jetPt_;
+  double jetEta_;
+  double pitchX_ = 0.01;              //100 um (pixel pitch in X)
+  double pitchY_ = 0.015;             //150 um (pixel pitch in Y)
+  static constexpr int jetDimX = 30;  //pixel dimension of NN window on layer2
+  static constexpr int jetDimY = 30;  //pixel dimension of NN window on layer2
+  static constexpr int Nlayer = 4;    //Number of layer used in DeepCore
+  static constexpr int Nover = 3;     //Max number of tracks recorded per pixel
+  static constexpr int Npar = 5;      //Number of track parameter
+
+private:
+  void beginJob();
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endJob();
+
+  // ----------member data ---------------------------
+  std::string propagatorName_;
+  edm::ESHandle<MagneticField> magfield_;
+  edm::ESHandle<GlobalTrackingGeometry> geometry_;
+  edm::ESHandle<Propagator> propagator_;
+
+  edm::EDGetTokenT<std::vector<reco::Vertex>> vertices_;
+  edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster>> pixelClusters_;
+  std::vector<SiPixelClusterWithTracks> allSiPixelClusters_;
```

Comment:
you are not using `allSiPixelClusters_` anywhere as far as I can tell, it's just cleared and reserved, but never filled.

Reply:
ok at this point seems I can remove completely also the struct of the class? 
https://github.com/cms-sw/cmssw/pull/32222/files/a9800bb408c991ba2e5f75e0d9ce8084203fe67f#diff-cdba9f2f7353b9cefac6050adf2a1065ebfabf6799dbbe0cedfd9eeae9949cb1R79-R94

Reply:
yes, if you're not using them, they should be removed.